### PR TITLE
Bug Fix : Dates in Events Page closes #1590

### LINKED ISSUE #1590: Add error handling for the start and end dates in the edit event page
**Describe the bug**
The start and end dates in the edit event page have no error handling with regards to the date chosen from the mobile application frontend.

Follow the steps outlined here till 8:18
[Follow the steps outlined here till the 50 second mark](https://www.youtube.com/watch?v=ix3eSReJj2k?t=446)

**To Reproduce**

Screenshot at Feb 27 08-05-51

**Expected behavior**
A notification should be given if you choose an impossible start and end date.

**Actual behavior**
Nothing

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by SiddheshKukade**
**What kind of change does this PR introduce?**

Bug fix in the app

**Issue Number:**

Fixes #1590

**Did you add tests for your changes?**
Not required

**Snapshots/Videos:**

https://user-images.githubusercontent.com/65951872/222955844-9e0ba496-3610-4226-9690-92fde11b58db.mp4

**If relevant, did you update the documentation?**

Not required
**Summary**
Fixed the date bug.

**Does this PR introduce a breaking change?**
No
 **Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

**Comment by Cioppolo14** [comment]
@SiddheshKukade Please fix the failing tests.

File: lib/views/after_auth_screens/events/create_event_page.dart [hunk 1]
```diff
@@ -183,18 +186,36 @@ class _CreateEventPageState extends State<CreateEventPage> {
...
                         );
-
-                        setState(() {
-                          model.eventEndTime = time;
-                        });
+                        final currTimeToInt = time.hour + time.minute / 60.0;
+                        final startTime = model.eventStartTime;
+                        final startTimeToInt =
+                            startTime.hour + startTime.minute / 60;
+                        if (startTimeToInt.compareTo(currTimeToInt) < 0) {
```
Comment (TheHazeEffect) [inline lib/views/after_auth_screens/events/create_event_page.dart:209]: Just double checking this
would this stop a user from setting event time to 5:30pm on Tuesday when it's currently 6:30pm on Sunday?

**Comment by palisadoes** [comment]
@SiddheshKukade  Please fix the failing tests

**Comment by SiddheshKukade** [comment]
> @SiddheshKukade  Please fix the failing tests

Yes sir, I'm on it.

File: lib/views/after_auth_screens/events/create_event_page.dart (hunk 1 above)
Comment (SiddheshKukade) [reply lib/views/after_auth_screens/events/create_event_page.dart]: Thanks for the review, 
You're right. Now I've added that restriction and also the check at the start date to make sure that start date is not any date before today's date.
Please take a look  : https://github.com/PalisadoesFoundation/talawa/pull/1616/commits/b7f368748c8ef9c53ee2b68f311dda7ff2fdba38

Comment (SiddheshKukade) [reply lib/views/after_auth_screens/events/create_event_page.dart]: Now I've tried all the combination of inputs and it is giving correct result with proper date and time. ✔️

**Comment by SiddheshKukade** [comment]
@palisadoes  @TheHazeEffect the errors are fixed and code is ready  to be reviewed again.

**Comment by palisadoes** [comment]
@SiddheshKukade There are some warnings about code coverage in the tests:
- https://github.com/PalisadoesFoundation/talawa/pull/1616/files

**Review by noman2002 — APPROVED** [review]

**Comment by SiddheshKukade** [comment]
> @SiddheshKukade There are some warnings about code coverage in the tests:

image


Sir, these are just checks stating that new lines are not covered in the current tests. The current coverage is almost 98%. I think we should wait untll the code coverage is below a threshold to create an issue to write tests  for this file.

image

**Pull request merged**